pkg/charts/additionalchart.go

```diff
@@ -48,22 +48,27 @@ func (c *AdditionalChart) ApplyMainChanges(pkgFs billy.Filesystem) error {
 	if err != nil {
 		return fmt.Errorf("encountered error while trying to get the main chart's working directory: %s", err)
 	}
-	if c.CRDChartOptions.UseTarArchive {
-		if err := helm.ArchiveCRDs(pkgFs, mainChartWorkingDir, path.ChartCRDDir, c.WorkingDir, path.ChartExtraFileDir); err != nil {
-			return fmt.Errorf("encountered error while trying to bundle and compress CRD files from the main chart: %s", err)
-		}
-	}
 	if err := helm.CopyCRDsFromChart(pkgFs, mainChartWorkingDir, path.ChartCRDDir, c.WorkingDir, c.CRDChartOptions.CRDDirectory); err != nil {
 		return fmt.Errorf("encountered error while trying to copy CRDs from %s to %s: %s", mainChartWorkingDir, c.WorkingDir, err)
 	}
-	if err := helm.DeleteCRDsFromChart(pkgFs, mainChartWorkingDir); err != nil {
-		return fmt.Errorf("encountered error while trying to delete CRDs from main chart: %s", err)
-	}
 	if c.CRDChartOptions.AddCRDValidationToMainChart {
 		if err := AddCRDValidationToChart(pkgFs, mainChartWorkingDir, c.WorkingDir, c.CRDChartOptions.CRDDirectory); err != nil {
 			return fmt.Errorf("encountered error while trying to add CRD validation to %s based on CRDs in %s: %s", mainChartWorkingDir, c.WorkingDir, err)
 		}
 	}
+	if c.CRDChartOptions.UseTarArchive {
```

Comment:
That was what I initially did, but looking at https://github.com/rancher/charts-build-scripts/pull/131#discussion_r1589631903 the code comes out somewhat convoluted and hard to read because of all the difference branches and cases we need to cover. Realistically the actual performance hit is pretty small, and given the choice between slightly slower and cleaner vs slightly faster and messy I'd choose cleaner.